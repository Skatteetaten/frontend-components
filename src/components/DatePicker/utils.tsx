export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
