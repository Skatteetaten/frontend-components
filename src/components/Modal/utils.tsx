import { useEffect } from 'react';

export const createModalDomPlacement = (
  setFlag: (a: boolean) => void,
  shadowRootNode: Document | ShadowRoot | undefined
) => {
  if (!shadowRootNode) {
    setFlag(true);
    return;
  }

  const styleInjectorWrapperId = 'style-injector';
  const modalWrapperId = 'modal-wrapper';

  if (shadowRootNode.getElementById(modalWrapperId)) {
    setFlag(true);
  } else {
    const modalWrapper = document.createElement('div');
    modalWrapper.id = modalWrapperId;
    const styleInjectorWrapper = shadowRootNode.getElementById(
      styleInjectorWrapperId
    );
    const parent = styleInjectorWrapper ? styleInjectorWrapper : shadowRootNode;
    parent.appendChild(modalWrapper);
    setFlag(true);
  }
};

export const useEscOnPress = (
  ref: HTMLDivElement | null,
  onClose?: (ref: HTMLDivElement) => void
): void => {
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent): void => {
      if (ref == null) {
        return;
      }
      if (event.code === 'Escape' && onClose) {
        onClose(ref);
      }
    };

    if (ref) {
      window.addEventListener('keydown', onKeyDown as EventListener, false);
    }
    return function cleanup(): void {
      if (window) {
        window.removeEventListener('keydown', onKeyDown as EventListener);
      }
    };
  }, [ref, onClose]);
};
