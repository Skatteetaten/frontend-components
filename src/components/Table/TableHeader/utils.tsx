import { TFunction } from 'react-i18next';

export const getIconNameForTheadCell = (
  isSorted: boolean,
  isSortedAscending: boolean
) => {
  if (isSorted) {
    return isSortedAscending ? 'ArrowDown' : 'ArrowUp';
  }

  return 'ArrowUpDown';
};

export const getAriaLabelForTheadCell = (
  name: string | JSX.Element,
  fieldName: string,
  isSorted: boolean,
  isSortedAscending: boolean,
  t: TFunction<Array<string>>
) => {
  let ariaLabel = fieldName;
  if (typeof name === 'string') {
    ariaLabel = isSorted
      ? isSortedAscending
        ? name.concat(' ', t('table.sorted_ascending'))
        : name.concat(' ', t('table.sorted_descending'))
      : name.concat(' ', t('table.sortable'));
  }

  return ariaLabel;
};
