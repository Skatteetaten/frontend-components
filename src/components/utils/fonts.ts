export enum FontSizes {
  mini = '10px',
  xSmall = '12px',
  small = '14px',
  smallPlus = '15px',
  medium = '16px',
  mediumPlus = '17px',
  icon = '20px',
  large = '18px',
  largePlus = '20px',
  xLarge = '22px',
  xxLarge = '28px',
  superLarge = '42px',
  mega = '72px'
}

export enum FontWeights {
  light = 100,
  semilight = 300,
  regular = 400,
  semibold = 600,
  bold = 700
}

export enum IconFontSizes {
  xSmall = '10px',
  small = '12px',
  medium = '16px',
  large = '20px',
  xlarge = '24px',
  xxlarge = '32px',
  mega = '40px'
}
// @ts-ignore TODO
export const createFont = (size, weight, localeCode = 'no') => {
  return {
    fontFamily: "'Helvetica Neue', Helvetica , Arial, sans-serif",
    MozOsxFontSmoothing: 'grayscale',
    WebkitFontSmoothing: 'antialiased',
    fontSize: size,
    fontWeight: weight
  };
};

export default {
  tiny: createFont(FontSizes.mini, FontWeights.semibold),
  xSmall: createFont(FontSizes.xSmall, FontWeights.regular),
  small: createFont(FontSizes.small, FontWeights.regular),
  smallPlus: createFont(FontSizes.smallPlus, FontWeights.regular),
  medium: createFont(FontSizes.medium, FontWeights.regular),
  mediumPlus: createFont(FontSizes.mediumPlus, FontWeights.regular),
  large: createFont(FontSizes.large, FontWeights.semilight),
  xLarge: createFont(FontSizes.xLarge, FontWeights.light),
  xxLarge: createFont(FontSizes.xxLarge, FontWeights.light),
  superLarge: createFont(FontSizes.superLarge, FontWeights.light),
  mega: createFont(FontSizes.mega, FontWeights.light)
};
