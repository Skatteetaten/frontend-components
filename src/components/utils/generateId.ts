export const generateId = () =>
  `${Math.floor(Math.random() * Math.floor(Number.MAX_SAFE_INTEGER))}`;
