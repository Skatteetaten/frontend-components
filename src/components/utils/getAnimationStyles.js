// import { keyframes } from '@uifabric/merge-styles'; //kan benyttes hvis man skal lage egne keyframe anismasjoner
import { AnimationClassNames } from '@uifabric/styling';

export const Animation = {
  errorMessage: AnimationClassNames.slideDownIn20
};
