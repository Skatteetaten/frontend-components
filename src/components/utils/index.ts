export { DefaultPalette as Palette, skeColor, skePalette } from './palette';
export type { PaletteProps, SkeColorProp } from './palette';
export {
  Fonts,
  FontSizes,
  LineHeightSizes,
  FontWeights,
  IconFontSizes,
  createFont,
} from './fonts';
export { omstrukturerFlatJSON, t } from './i18n/i18n';
export { getSrOnlyStyle } from './getSrOnlyStyle';
export { languages } from './i18n/languages';
export type { FontSizeType } from './fonts';
export { getFocusStyle } from './getFocusStyle';
export { Heading } from './Heading';
export { UseScreen } from './ScreenPlugin';
export { default as SkeIcons } from './icons/icons-ske';
export { Animation } from './getAnimationStyles';
export { takeIf } from './helpers';
export { generateId } from './generateId';
export { useHotkeys } from './useHotkeys';
