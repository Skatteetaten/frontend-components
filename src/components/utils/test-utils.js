import renderer from 'react-test-renderer';

export const matches = (children) =>
  expect(renderer.create(children).toJSON()).toMatchSnapshot();
