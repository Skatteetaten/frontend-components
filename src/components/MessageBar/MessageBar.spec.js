import React from 'react';
import { mount } from 'enzyme';
import { matches } from './../utils/test-utils';
import { MessageBar } from '../index';

function oppsettFullDOM(props) {
  return mount(
    <MessageBar {...props}>
      <p>Satsene for denne avgiftstypen ble oppdatert 01.01.2017.</p>
    </MessageBar>
  );
}

describe('MessageBar komponent', () => {
  it('matcher snapshot', () => {
    matches(<MessageBar />);
  });

  it('renderer MessageBar med default props ', () => {
    const wrapper = oppsettFullDOM({});

    expect(wrapper.prop('actions')).toEqual(undefined);
    expect(wrapper.prop('type')).toEqual(0);
    expect(wrapper.prop('isMultiline')).toEqual(true);
    expect(wrapper.prop('onClick')).toEqual(undefined);
  });

  it('renderer "feil" MessageBar  ', () => {
    const wrapper = oppsettFullDOM({
      type: MessageBar.Type.error,
    });

    const icon = wrapper.find('IconBase');
    expect(icon.prop('iconName')).toEqual('ErrorBadge');
    expect(wrapper.prop('type')).toEqual(1);
  });

  it('renderer "varsel" MessageBar  ', () => {
    const wrapper = oppsettFullDOM({
      type: MessageBar.Type.warning,
    });

    const icon = wrapper.find('IconBase');
    expect(icon.prop('iconName')).toEqual('Info');
    expect(wrapper.prop('type')).toEqual(5);
  });

  it('renderer "strengt fortrolig" MessageBar  ', () => {
    const wrapper = oppsettFullDOM({
      type: MessageBar.Type.severeWarning,
    });

    const icon = wrapper.find('IconBase');
    expect(icon.prop('iconName')).toEqual('Warning');
    expect(wrapper.prop('type')).toEqual(3);
  });

  it('renderer "ok/positiv" MessageBar  ', () => {
    const wrapper = oppsettFullDOM({
      type: MessageBar.Type.success,
    });

    const icon = wrapper.find('IconBase');
    expect(icon.prop('iconName')).toEqual('Completed');
    expect(wrapper.prop('type')).toEqual(4);
  });

  it('renderer MessageBar med riktig props  ', () => {
    const wrapper = oppsettFullDOM({
      id: 'messagebar-id',
      className: 'messagebar-class',
      isMultiline: false,
    });

    expect(wrapper.prop('id')).toEqual('messagebar-id');
    expect(wrapper.prop('className')).toEqual('messagebar-class');
    expect(wrapper.exists('.ms-MessageBar-singleline')).toEqual(true);
  });

  it('renderer MessageBar på nytt når knapp trykkes ved bruk av "onRenderAfterDuration"', () => {
    //TODO: Teste at messagbar rendrer på nytt ved bruk av onRenderAfterDuration
  });
});
