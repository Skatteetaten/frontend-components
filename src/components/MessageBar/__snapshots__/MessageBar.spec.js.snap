// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBar komponent matcher snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "background": "#ffffff",
      }
    }
  >
    <div
      className="ms-MessageBar ms-MessageBar-multiline entered root-42"
    >
      <div
        className="ms-MessageBar-content content-43"
      >
        <div
          aria-hidden={true}
          className="ms-MessageBar-icon iconContainer-44"
        >
          <i
            aria-hidden={true}
            className="icon-54"
            data-icon-name="Info"
          >
            ï‹¼
          </i>
        </div>
        <div
          aria-live="polite"
          className="ms-MessageBar-text text-46"
          id="MessageBar0"
          role="status"
        >
          <span
            className="ms-MessageBar-innerText innerText-47"
            role="status"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
